mbuild/box.py

```diff
@@ -3,179 +3,414 @@
+
+        Based on the box vectors, return the parameters to describe the box in
+        terms of the Bravais lattice parameters:
+            a,b,c = the edges of the Box
+            alpha, beta, gamma = angles(tilt) of the parallelepiped, in degrees
+
+        Returns
+        -------
+        parameters : tuple of floats,
+            (a, b, c, alpha, beta, gamma)
+        """
+        (alpha, beta, gamma) = self.angles
+        (Lx, Ly, Lz) = self.lengths
+        return Lx, Ly, Lz, alpha, beta, gamma
+
+    def __repr__(self):
+        """Return a string representation of the box."""
+        (Lx, Ly, Lz, xy, xz, yz) = self.box_parameters
+        format_precision = f".{self._precision}f" if self._precision else ""
+        desc = (
+            f"Box: Lx={Lx:{format_precision}}, "
+            f"Ly={Ly:{format_precision}}, "
+            f"Lz={Lz:{format_precision}}, "
+            f"xy={xy:{format_precision}}, "
+            f"xz={xz:{format_precision}}, "
+            f"yz={yz:{format_precision}}, "
         )
-        self._mins = _BoxArray(
-            array=(self.mins - (0.5 * lengths - 0.5 * self.lengths)),
-            var="mins",
-            box=self,
+        return desc
+
+    def _from_vecs_to_lengths_tilt_factors(self):
+        # vectors should already be aligned by _normalize_box
+        v1 = self._vectors[0, :]
+        v2 = self._vectors[1, :]
+        v3 = self._vectors[2, :]
+
+        Lx = np.linalg.norm(v1)
+        Ly = np.linalg.norm(v2)
+        Lz = np.linalg.norm(v3)
+
+        v1_dot_v2 = np.dot(v1, v2)
+        v1_dot_v3 = np.dot(v1, v3)
+        v2_dot_v3 = np.dot(v2, v3)
+
+        xy = v1_dot_v2 / (Lx * Ly)
+        xz = v1_dot_v3 / (Lx * Lz)
+        yz = (v2_dot_v3 - ((v1_dot_v2 / Lx) * (v1_dot_v3 / Lx))) / (Ly * Lz)
+
+        return Lx, Ly, Lz, xy, xz, yz
+
+    def _get_angles(self):
+        return _calc_angles(self.vectors)
+
+
+def _validate_box_vectors(box_vectors):
+    """Determine if the vectors are in the convention we use.
+
+    This method will parse the provided box vectors, determine if the vectors
+    follow the conventions the Box class adheres to. In this case:
+
+    1. It is a 3x3 matrix that can be coerced into a numpy array of floats
+    2. Vectors are in a right-handed basis (determinant of matrix is +)
+    3. The first vector is aligned along the [1,0,0] direction
+    4. The second vector in aligned along the xy plane
+    5. The third vector can align freely in the x,y, and +z direction
+
+    If the vectors are not right-handed, a warning will be raised, and the
+    vectors will be transformed into the right-handed coordinate system.
+
+    If the three vectors are not following conventions 3-5, the matrix will be
+    transformed to comply with them, and will also raise a warning.
+    """
+    vecs = np.asarray(box_vectors, dtype=np.float64)
+    vecs.reshape(3, 3)
+
+    return _normalize_box(vecs)
+
+
+def _lengths_angles_to_vectors(lengths, angles, precision):
```

Comment:
This precision value works.  I think we should make it an option, like precision=8 to 12 as a default, and let the user set it to whatever.  This would be my preference for all the precision arguments, however, I can live with this "as is", as this may be ultra picky.